sklearn/model_selection/_search.py

```diff
@@ -648,6 +648,35 @@ def fit(self, X, y=None, groups=None, **fit_params):
             all_candidate_params = []
             all_out = []
 
+            def is_none(x):
+                return x is None
+
+            def collape_nones(xs):
+                return None if xs is None or any(map(is_none, xs)) else xs
```

Comment:
Perhaps the function can be written like with a generator instead of a map?

```python
def collapse_nones_g(xs):
    if xs is None or any(x is None for x in xs):
        return None
    return xs
```

Reply:
It's already a generator:

```
In [32]: %time sum(range(200000000))
CPU times: user 3.86 s, sys: 9.2 ms, total: 3.86 s
Wall time: 3.86 s
```

```
In [34]: %timeit any(map(lambda x: x > 2, range(200000000)))
940 ns ± 6.97 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```